pdns/lua-record.cc

```diff
@@ -40,6 +40,15 @@ using wiplist_t = std::unordered_map<int, string>;
 using ipunitlist_t = vector<pair<int, iplist_t> >;
 using opts_t = std::unordered_map<string,string>;
 
+unsigned int pdns_stou_nonzero(const std::string& str, size_t * idx = 0, int base = 10)
```

Comment:
Maybe this could be useful to leave it in `misc.cc` ?

Reply:
I was unsure. If you also feel that way, let's do it :)

Reply:
is there a reason we cannot use boost::lexical_cast instead? do we really need non-base10?

Reply:
Ah, makes sense. Of course it's weird we're parsing at all - it was a number before we pulled it out of Lua as a string!